Source/OxyPlot/Series/LineSeries.cs

```diff
@@ -672,25 +672,70 @@ protected void RenderLegendOnLine(IRenderContext rc)
             // Find the position
             DataPoint point;
             var ha = HorizontalAlignment.Left;
-            double dx;
+            var va = VerticalAlignment.Middle;
+            double dx = 0;
+            double dy = 0;
+            double rotation = 0;
+            DataPoint point2;
             switch (this.LineLegendPosition)
             {
                 case LineLegendPosition.Start:
-
                     // start position
                     point = this.ActualPoints[0];
                     ha = HorizontalAlignment.Right;
                     dx = -4;
                     break;
-                default:
+                case LineLegendPosition.StartInside:
 
+                    // start position
+                    point = this.ActualPoints[0];
+                    ha = HorizontalAlignment.Left;
+                    va = VerticalAlignment.Bottom;
+                    dx = 4;
+                    break;
+                case LineLegendPosition.HalfWayAbove:
+                    point = this.ActualPoints[this.ActualPoints.Count / 2];
+                    dx = 0;
+                    dy = -4;
+                    ha = HorizontalAlignment.Center;
+                    va = VerticalAlignment.Bottom;
+                    break;
+                case LineLegendPosition.HalfWayAboveSloped:
+                    point = this.ActualPoints[this.ActualPoints.Count / 2];
+                    point2 = this.ActualPoints[this.ActualPoints.Count / 2 - 1];
+                    rotation = Math.Atan2(point.y - point2.y, point.x - point2.x) * -360 / Math.PI;
+                    dx = 0;
+                    dy = -4;
+                    ha = HorizontalAlignment.Center;
+                    va = VerticalAlignment.Bottom;
+                    break;
+                case LineLegendPosition.HalfWayBelow:
+                    point = this.ActualPoints[this.ActualPoints.Count / 2 - 1];
+                    dx = 0;
+                    dy = 4;
+                    ha = HorizontalAlignment.Center;
+                    va = VerticalAlignment.Top;
+                    break;
+                case LineLegendPosition.HalfWayBelowSloped:
+                    point = this.ActualPoints[this.ActualPoints.Count / 2];
+                    point2 = this.ActualPoints[this.ActualPoints.Count / 2 - 1];
+                    rotation = Math.Atan2(point.y - point2.y, point.x - point2.x) * -360 / Math.PI;
+                    dx = 0;
+                    dy = 4;
+                    ha = HorizontalAlignment.Center;
+                    va = VerticalAlignment.Top;
+                    break;
+                case LineLegendPosition.End:
                     // end position
+                    dy = 4;
                     point = this.ActualPoints[this.ActualPoints.Count - 1];
-                    dx = 4;
                     break;
+                default:
+                    //was end previously
```

Comment:
Note: this is a breaking change.

Reply:
I removed the comment. Originally the End case would go to the default as if it was an if else statement. I added the end case so there should be no backwards compatibility issues.